test/unit/access-denied-test.js

```diff
@@ -0,0 +1,353 @@
+  t.ok(result, 'Alice should have Append access implied by Write access')
+  t.end()
+})
+
+// Straight ACL access test
+test('acl-check accessDenied() test - accessTo', function (t) {
+  let container = $rdf.sym('https://alice.example.com/docs/')
+  let containerAclUrl = 'https://alice.example.com/docs/.acl'
+  let containerAcl = $rdf.sym(containerAclUrl)
+
+  const store = $rdf.graph() // Quad store
+  const ACLtext = prefixes +
+  ` <#auth> a acl:Authorization;
+    acl:mode acl:Read, acl:Write;
+    acl:agent alice:me;
+    acl:accessTo <${container.uri}> .
+  `
+  $rdf.parse(ACLtext, store, containerAclUrl, 'text/turtle')
+
+  var result = aclLogic.accessDenied(store, container, null, containerAcl, bob, [ ACL('Write')])
+  t.ok(result, 'Bob Should not have access')
+  t.equal(result, 'User Unauthorized', 'Correct reason')
+
+  t.end()
+})
+
+// Inheriting permissions from directory defaults
+test('acl-check accessDenied() test - default/inherited', function (t) {
+  let container = $rdf.sym('https://alice.example.com/docs/')
+  let containerAcl = $rdf.sym('https://alice.example.com/docs/.acl')
+  let file1 = $rdf.sym('https://alice.example.com/docs/file1')
+  let file2 = $rdf.sym('https://alice.example.com/docs/stuff/file2')
+  var result
+  const store = $rdf.graph()
+  /*
+  let ACLtext = prefixes + ` <#auth> a acl:Authorization;
+    acl:mode acl:Read;
+    acl:agent bob:me;
+    acl:accessTo <${file1.uri}> .
+`
+  $rdf.parse(ACLtext, store, containerAcl.uri, 'text/turtle')
+*/
+  let containerAclText = prefixes + ` <#auth> a acl:Authorization;
+      acl:mode acl:Read;
+      acl:agent alice:me;
+      acl:default <${container.uri}> .
+`
+  $rdf.parse(containerAclText, store, containerAcl.uri, 'text/turtle')
+
+  result = !aclLogic.accessDenied(store, file1, container, containerAcl, alice, [ ACL('Read')])
+  t.ok(result, 'Alice should have Read acces inherited')
+
+  result = !aclLogic.accessDenied(store, file2, container, containerAcl, alice, [ ACL('Read')])
+  t.ok(result, 'Alice should have Read acces inherited 2')
+
+  result = aclLogic.accessDenied(store, file2, container, containerAcl, alice, [ ACL('Write')])
+  t.ok(result, 'Alice should NOT have Write acces inherited')
+
+  t.end()
+})
+
+
+
+// Inheriting permissions from directory defaults
+test('acl-check accessDenied() test - default/inherited', function (t) {
+  let container = $rdf.sym('https://alice.example.com/docs/')
+  let containerAcl = $rdf.sym('https://alice.example.com/docs/.acl')
+  let file1 = $rdf.sym('https://alice.example.com/docs/file1')
+  let file2 = $rdf.sym('https://alice.example.com/docs/stuff/file2')
+  var result
+  const store = $rdf.graph()
+  /*
+  let ACLtext = prefixes + ` <#auth> a acl:Authorization;
+    acl:mode acl:Read;
+    acl:agent bob:me;
+    acl:accessTo <${file1.uri}> .
+    `
+  $rdf.parse(ACLtext, store, containerAcl.uri, 'text/turtle')
+*/
```

Comment:
commented?

Reply:
Yeah, actually, many of the tests were taken from the check-access-test, and this was commented out there. I guess I should remove the outcommented stuff.